Description: Don't wipe generated iface target in active domains
 .
 Wipe generated interface target only when reading configuration of
 inactive domains.
Author: Jiri Denemark <jdenemar@redhat.com>
Origin: http://libvirt.org/git/?p=libvirt.git;a=commit;h=362bc09a4d9441f9abd14956546777f86d8b9d79 (adapted)
Applied-Upstream: commit:362bc09a4d9441f9abd14956546777f86d8b9d79 (adapted)
Author: Jiri Denemark <jdenemar@redhat.com> 
Bug: https://bugzilla.redhat.com/show_bug.cgi?id=591438 
Bug-Ubuntu: https://launchpad.net/bugs/668042 
Last-Update: 2010-11-25

--- a/src/conf/domain_conf.c
+++ b/src/conf/domain_conf.c
@@ -1155,7 +1155,8 @@
                        xmlStrEqual(cur->name, BAD_CAST "target")) {
                 ifname = virXMLPropString(cur, "dev");
                 if ((ifname != NULL) &&
-                    (STRPREFIX((const char*)ifname, "vnet"))) {
+                    ((flags & VIR_DOMAIN_XML_INACTIVE) &&
+                      (STRPREFIX((const char*)ifname, "vnet")))) {
                     /* An auto-generated target name, blank it out */
                     VIR_FREE(ifname);
                 }
